libraries/AP_GPS/AP_GPS_SBP2.cpp

```diff
@@ -476,7 +481,35 @@ AP_GPS_SBP2::logging_log_raw_sbp(uint16_t msg_type,
             time_us         : time_us,
             msg_type        : msg_type,
         };
-        memcpy(pkt2.data2, &msg_buff[64], MIN(msg_len - 64,192));
+        memcpy(pkt2.data2, &msg_buff[64], MIN(msg_len - 64,104));
         gps._DataFlash->WriteBlock(&pkt2, sizeof(pkt2));
+
+        if (msg_len > 168) {
+            struct log_SbpRAW2 pkt3 = {
+                LOG_PACKET_HEADER_INIT(LOG_MSG_SBPRAW2),
```

Comment:
I think this needs to be a different third type or it could get messy correlating all the messages.